Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -88,8 +193,74 @@ public static int[] InsertLast(int[] array, int number)
         /// <returns>A new array with element inserted at a given index. If an array is empty or null, returns input array.</returns>
         public static int[] InsertAt(int[] array, int number, int index)
         {
-            // ToDo: implement.
-            return array;
+
+            if (!IsValidCaseForInsertForNullOrEmpty(array, index))
+            {
+                return array;
+            }
+
+            if (!isValidCaseForInsertForNullOrEmpty(array, number, index)  )
+            {
+                if (index == 0)
+                {
+                    return new int[] { number };
+                }
+                return array; 
+            }
+
+           
+            
+            var newArray = Insert(array, number, index);
+
+            return newArray;
+
+        }
+
+        public static bool IsValidCaseForInsertForNullOrEmpty(int[] array, int index)
+        {
+            if (index >= 0)
+            {
+                return true;  
+            }
+            return false; 
+        } 
+           
+        
+            
+
+        public static bool isValidCaseForInsertForNullOrEmpty(int[] array, int number, int index)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Rename method 'isValidCaseForInsertForNullOrEmpty' to match pascal case naming rules, consider using 'IsValidCaseForInsertForNullOrEmpty'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887669)

Reply:
solved